lessons/advanced/strings-in-depth.md

```diff
@@ -0,0 +1,156 @@
+---
+layout: page
+title: Strings in-depth 
```

Comment:
Great!  @setzer777 let's add this as the last lesson in the Basics and rename it to "Strings", then I think it's good to go 👍